src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -288,21 +365,28 @@ namespace ts.server.typingsInstaller {
                         this.log.writeLine(`Installed typings ${JSON.stringify(scopedTypings)}`);
                     }
                     const installedTypingFiles: string[] = [];
+                    const typesPackageName = (packageName: string) => `@types/${packageName}`;
                     for (const packageName of filteredTypings) {
                         const typingFile = typingToFileName(cachePath, packageName, this.installTypingHost, this.log);
                         if (!typingFile) {
                             this.missingTypingsSet.set(packageName, true);
                             continue;
                         }
-                        if (!this.packageNameToTypingLocation.has(packageName)) {
-                            this.packageNameToTypingLocation.set(packageName, typingFile);
-                        }
+
+                        const newTimestamp = Date.now();
+                        const newTyping: JsTyping.CachedTyping = { typingLocation: typingFile, timestamp: newTimestamp };
+                        this.packageNameToTypingLocation.set(packageName, newTyping);
+                        this.typeDeclarationTimestamps[typesPackageName(packageName)] = newTimestamp;
                         installedTypingFiles.push(typingFile);
                     }
                     if (this.log.isEnabled()) {
                         this.log.writeLine(`Installed typing files ${JSON.stringify(installedTypingFiles)}`);
                     }
 
+                    const newFileContents: TypeDeclarationTimestampFile = { entries: this.typeDeclarationTimestamps };
+                    const timestampJson = combinePaths(cachePath, timestampsFile);

```

Comment:
Refactor so there's exactly one place which computes this file path (currently duplicated at line 228)

Reply:
Moved. We can't only compute this in exactly one place since it can sometimes be different. That is, each request to install types can provide its own path that can differ from the one provided on creation of the `TypingsInstaller`.